### PR TITLE
feat: change statefulset to deployment

### PR DESCRIPTION
**Pull request by brianbinbin**

### CODE REVIEW

File: deploy/chart/kuiper/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+          items:
+            - key: {{ ternary "tls.crt" "ca.crt" .Values.tls.existingSecretFullChain }}
+              path: xyz-rootca.pem
+            - key: tls.crt
+              path: xyz-certificate.pem
+            - key: tls.key
+              path: xyz-private.pem.key
+        {{- end }}
+      containers:
+        - name: kuiper
```
Comment (Rory-Z) [inline deploy/chart/kuiper/templates/deployment.yaml:46]: Maybe `ekuiper` ?

File: deploy/chart/kuiper/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+          image: "{{ .Values.image.repository }}:{{ .Chart.AppVersion }}-slim"
```
Comment (Rory-Z) [inline deploy/chart/kuiper/templates/deployment.yaml:47]:
```suggestion
          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}-slim"
```

Comment (Rory-Z) [reply deploy/chart/kuiper/templates/deployment.yaml:47]: I'm not sure we need to hardcode the slim image ?

File: deploy/chart/kuiper/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          {{- if .Values.kuiperEnv.enabled }}
```
Comment (Rory-Z) [inline deploy/chart/kuiper/templates/deployment.yaml:49]: Maybe `ekuiperEnv` ?

File: deploy/chart/kuiper/values.yaml [hunk 4]
```diff
@@ -2,25 +2,32 @@
 # This is a YAML-formatted file.
 # Declare variables to be passed into your templates.
 
+# -- only support one replica
 replicaCount: 1
```
Comment (Rory-Z) [inline deploy/chart/kuiper/values.yaml:6]: Maybe we can hardcode replicaCount=1 in deployment

File: deploy/chart/kuiper/values.yaml [hunk 5]
```diff
@@ -150,188 +170,12 @@ kuiperConfig:
...
-    dedup:
-      interval: 100
-      deduplicate: 50
-  "sources/zmq.yaml":
-    #Global Zmq configurations
-    default:
-      server: tcp://127.0.0.1:5563
+# TODO
+# -- remove this when the mqtt_source configmap is available
+kuiperEnv:
```
Comment (Rory-Z) [inline deploy/chart/kuiper/values.yaml:176]: Maybe ekuiperEnv ?

**Review by Rory-Z — CHANGES_REQUESTED** [review]
Please remove `.vscode` path and add this to `.gitignore`

File: deploy/chart/kuiper/templates/deployment.yaml (hunk 2 above)
Comment (brianbinbin) [reply deploy/chart/kuiper/templates/deployment.yaml:47]: Jiyong suggested that we'd better use slim.

**Comment by Rory-Z** [comment]
Please fix failed check

File: deploy/chart/ekuiper/README.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+```
+git clone https://github.com/lf-edge/ekuiper.git
+cd deploy/chart/ekuiper
+helm install my-ekuiper .
+```
+
+- From Helm repo
+```
+helm repo add emqx https://repos.emqx.io/charts
+helm install my-ekuiper emqx/kuiper
```
Comment (Rory-Z) [inline deploy/chart/ekuiper/README.md:18]: Do we need update repo name to `emqx/ekuiper` ? @ngjaying

Comment (ngjaying) [reply deploy/chart/ekuiper/README.md:18]: yes

Comment (ngjaying) [reply deploy/chart/ekuiper/README.md:18]: The docker image is lfedge/ekuiper, shall we rename to match that

File: deploy/chart/ekuiper/Chart.yaml [hunk 7]
```diff
@@ -14,8 +14,8 @@ type: application
 
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
-version: 1.2.0
+version: 1.3.0
 
 # This is the version number of the application being deployed. This version number should be
 # incremented each time you make changes to the application.
-appVersion: 1.2.0
+appVersion: 1.5.0
```
Comment (ngjaying) [inline deploy/chart/ekuiper/Chart.yaml:21]: The latest is 1.5.1, shall we change to it? Do we need to change here for each release?

**Review by ngjaying — APPROVED** [review]

**Review by ngjaying — APPROVED** [review]

**Pull request merged**